src/__internal__/focus-trap/focus-trap.spec.js

```diff
@@ -34,32 +37,78 @@ describe("FocusTrap", () => {
   });
   const otherKey = new KeyboardEvent("keydown", { keyCode: 32 });
 
-  describe("when autoFocus is false", () => {
-    let wrapper;
+  describe("triggerRefocusFlag", () => {
+    afterEach(() => {
+      wrapper.unmount();
+    });
+
+    it("refocuses the last element that had focus within the trap when flag is set", () => {
+      const focusMock = jest.fn();
 
-    beforeEach(() => {
       wrapper = mount(
-        <MockComponent autoFocus={false}>
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
           <button type="button">Test button One</button>
-          <input type="text" />
+          <input onFocus={focusMock} type="text" />
         </MockComponent>,
         { attachTo: htmlElement }
       );
+      act(() => {
+        document.querySelectorAll("input")[0].focus();
+      });
+      act(() => {
+        document.querySelectorAll("input")[0].blur();
+      });
+      act(() => {
+        wrapper.setProps({ triggerRefocusFlag: true });
+      });
+      wrapper.update();
+      expect(focusMock).toHaveBeenCalledTimes(2);
```

Comment:
We should be testing that the same input that was focused re-received focus here

Reply:
That is what's being tested here, the same input is focused as the same callback is called twice